src/ddnnf/anomalies/atomic_sets.rs

```diff
@@ -271,6 +276,15 @@ impl Ddnnf {
     }
 }
 
+// removes inverse atomic sets
+fn sort_and_clean_atomicsets(atomic_sets: &mut Vec<Vec<i16>>) {
+    for atomic in atomic_sets.iter_mut() {
+        atomic.sort_by(|a, b| a.abs().cmp(&b.abs()));
+    }
+    atomic_sets.sort_unstable_by(|a, b| a[0].abs().cmp(&b[0].abs()).then(a[0].cmp(&b[0])));
+    atomic_sets.dedup_by(|a, b| a[0].abs() == b[0].abs());
```

Comment:
I think it should be faster to dedup first and then sort so that the sorting has less work to do.